### PR TITLE
Improve MembershipService tests

### PR DESCRIPTION
**Pull request by benjaminapetersen**
My first round of tests for `MembershipService` were rather terrible.  This is a bit better.
@jwforres @spadgett

### CODE REVIEW

File: test/spec/services/membership/membershipSpec.js [hunk 1]
```diff
@@ -169,19 +186,55 @@ describe('MembershipService', function() {
   });
 
   describe('#getSubjectKinds', function() {
-    // No test needed, this returns static data
+    it('should return the kind list in the preferred ourder', function() {
+      var kinds = MembershipService.getSubjectKinds();
```
Comment (jwforres) [inline test/spec/services/membership/membershipSpec.js]: If this getSubjectKinds is returning a map, there is no guarantee of iteration order. If you are expecting stable order that method probably needs to change.  If its returning an array and not a map I'm not sure why the next line is needed.

Comment (benjaminapetersen) [inline test/spec/services/membership/membershipSpec.js]: Oop, my test is actually wrong, this function isn't supposed to care about sort order.  The 'mapRolebindingsForUI' handles that.

**Comment by benjaminapetersen** [comment]
& rebased

File: test/spec/services/membership/membershipSpec.js [hunk 2]
```diff
@@ -169,19 +186,64 @@ describe('MembershipService', function() {
   });
 
   describe('#getSubjectKinds', function() {
-    // No test needed, this returns static data
+    it('should return the kind list in the preferred ourder', function() {
```
Comment (jwforres) [inline test/spec/services/membership/membershipSpec.js]: typo, order

Comment (benjaminapetersen) [inline test/spec/services/membership/membershipSpec.js]: :+1:

Comment (benjaminapetersen) [inline test/spec/services/membership/membershipSpec.js]: fixed

**Comment by benjaminapetersen** [comment]
@jwforres let me know if you think anything else needs to happen here.  I think these are definitely better than the previous round for testing membership.

**Comment by jwforres** [comment]
[merge]

**Pull request merged**